woocommerce/woocommerce
# Add WooCommerce > Subscriptions admin menu item and empty state offer screen

### All Submissions:






### Changes proposed in this Pull Request:



**Internal project:** pdjTHR-Vi-p2

This PR adds a Subscriptions item to the WooCommerce admin menu for eligible stores * . This admin screen is part of an experiment to offer eligible merchants WooCommerce Payments built in Subscriptions functionality. 

\* eligible stores are stores without any subscriptions functionality plugin installed, are older than 6 months and have at least 1 sale in the last 30 days. 

<p align="center">
<img width="1161" alt="Screen Shot 2022-05-11 at 2 45 09 pm" src="https://user-images.githubusercontent.com/8490476/167770904-c712e886-1457-4770-9a46-f1bd83d180a3.png">
</p>

Closes #32682
Closes #32691
Closes #32694

This PR is made up of the following PRs:

- https://github.com/woocommerce/woocommerce/pull/32796
- https://github.com/woocommerce/woocommerce/pull/32852
- https://github.com/woocommerce/woocommerce/pull/32875
- https://github.com/woocommerce/woocommerce/pull/32913
- https://github.com/woocommerce/woocommerce/pull/32867
- https://github.com/woocommerce/woocommerce/pull/32935
- https://github.com/woocommerce/woocommerce/pull/32958

### How to test the changes in this Pull Request:

1. Checkout this branch and build assets by running `pnpm nx watch woocommerce-admin` _**or**_ grab a zip of this PR here → [woocommerce.zip](https://github.com/woocommerce/woocommerce/files/8705383/woocommerce.zip)
2. **Plugin eligibility**: Ensure the following plugins are not installed at all: 
    - WooCommerce Subscriptions
    - WooCommerce Payments
    - Sumo Subscriptions
    - Yith Subscriptions
    - Subscripto
    - Subscriptions For WooCommerce
3. **Store tracking eligibility:** Your store allows tracking.
    - This checkbox should be checked: **WooCommerce → Settings → Advanced → WooCommerce.com → Enable tracking**.
3. **Store location eligibility:** For the menu item to be displayed your store needs to based in the US. 
     - Go to **WooCommerce > Settings** and make sure your store is set to a US based address. 
3. **Store age criteria:** To view the subscription menu item your store will need to have been active for more than 6 months - we're testing this experiment on established stores. If your store doesn't meet this criteria, you can use the following code snippet: 
    - `update_option( 'woocommerce_admin_install_timestamp', strtotime( '-6 month' ) );`
    - **_or_** change the `woocommerce_admin_install_timestamp` option to `1620866357` (timestamp from ~1 year ago) via the `/wp-admin/options.php` page.
5. **Paid order criteria:** Only stores with at least 1 order in the last 30 days are eligible. If your store doesn't have an order in the last 30 days. Either: 
     1. Set any existing pending or failed order to processing or completed 
     2. If you don't have any order at all, you can just manually create one in **WooCommerce > Orders > Add new** and set it to processing or completed status
     3. ⚠️ **Note:** this order sale query is cached and so you may need to delete the `'woocommerce-wcpay-subscriptions_recent_sales_eligibility'` transient if this criteria failed previously. 
6. If you meet all the criteria above you should see a **WooCommerce > Subscriptions (new)** menu item.
7. View the **WooCommerce > Subscriptions (new)** page. 
     - The 'new' badge should disappear on subsequent page loads.

<p align="center">
<img width="322" alt="Screen Shot 2022-05-11 at 2 34 56 pm" src="https://user-images.githubusercontent.com/8490476/167772790-ef299bac-7d6b-4245-a640-5022c8c23e35.png">
</p>

#### The error scenario

If an error occurs during plugin installation and activation an error message will be displayed directing the user to install the WooCommerce Payments plugin manually. 

To test this flow:

1. Add the following code snippet to your site. 
2. Go to **WooCommerce -> Subscriptions**
4. Click **"Get started"**
5. An error should occur installing the plugin. See example below. 

```
define( 'DISALLOW_FILE_MODS', true );
```

<p align="center">
<img width="1200" alt="Screen Shot 2022-05-11 at 3 00 59 pm" src="https://user-images.githubusercontent.com/8490476/167772680-233c53f2-5871-4f29-b7e3-3a66d5a5ca75.png">
</p>

❗**Note:** The link takes the user to the [WooCommerce Payments wp.org page](https://wordpress.org/plugins/woocommerce-payments/).

#### The success scenario

1. Go to **WooCommerce > Subscriptions**
2. Click **"Get started"**
3. WooCommerce Payments will be installed and activated in the background
4. Once complete, you will be directed to the “Add new product” page or to the WCPay onboarding flow depending on which experiment treatment you were assigned to. Please see the "Testing the treatments" section below for more instructions.

#### The dismiss scenario

1. Go to **WooCommerce > Subscriptions**
2. Click **"No thanks"**
3. You should be redirected to the **WooCommerce > Home** screen and the subscriptions menu item should no longer be displayed.

**Note:** If you need to bring the menu item back, delete or change the `woocommerce-wcpay-subscriptions_dismissed` wp option.

#### Testing the treatments

Download, install and activate the [WooCommerce Admin Test Helper](https://github.com/woocommerce/woocommerce-admin-test-helper) plugin.

Navigate to **Tools → WCA Test Helper → Experiments**. This page should contain a row for the "woocommerce_wcpay_subscriptions_page_202207_v1" experiment.

Here you can toggle between the control (treatment A) and the treatment (treatment B).

#### Treatment A

1. Toggle to the control experience from **Tools → WCA Test Helper → Experiments**.
2. Navigate to **WooCommerce → Subscriptions**.
3. Observe the following step headers in the "You’re only steps away from selling subscriptions" section.
     1. Create a subscription
     2. Create and connect your account
     3. Provide a few business details
4. Click the "Get started" button.
5. The WooCommerce Payments plugin will be installed & activated and you will be redirected to the “Add new product” page.

#### Treatment B

1. Toggle to the treatment experience from **Tools → WCA Test Helper → Experiments**.
2. Navigate to **WooCommerce → Subscriptions**.
3. Observe the following step headers in the "You’re only steps away from selling subscriptions" section.
     1. Create and connect your account
     2. Provide a few business details
     3. Create subscriptions
4. Click the "Get started" button.
5. The WooCommerce Payments plugin will be installed & activated and you will be redirected to the WCPay onboarding flow.

--

<br class="Apple-interchange-newline">

### Other information:




### FOR PR REVIEWER ONLY:
